lessons/r/dplyrmagrittr/lesson.ipynb

```diff
@@ -0,0 +1,650 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "select(iris, starts_with('Sepal'), ends_with('Length')) %>% head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# renaming with select\n",
+    "\n",
+    "select(iris, SL = Sepal.Length, SW = Sepal.Width) %>% head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Combining `select` and `filter` using the pipe:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "class(select(iris, Sepal.Length)) # is a data frame"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "select(iris, Sepal.Length) %>%\n",
+    "filter(Sepal.Length > 7.0) # notice lack of first argument here"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# could be restructured as -\n",
+    "iris %>%\n",
+    "    select(Sepal.Length) %>%\n",
+    "    filter(Sepal.Length > 7.0)\n",
+    "\n",
+    "# almost reads like a recipe!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "iris %>%\n",
+    "    select(Sepal.Length) %>%\n",
+    "    filter(Sepal.Length > 7.0) %>%\n",
+    "    arrange(Sepal.Length)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "iris %>%\n",
+    "    select(Sepal.Length) %>%\n",
+    "    filter(Sepal.Length > 7.0) %>%\n",
+    "    arrange(desc(Sepal.Length))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "head(iris)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# arranging by x *then* y\n",
+    "\n",
+    "iris %>%\n",
+    "    select(Petal.Length:Species) %>%\n",
+    "    arrange(Petal.Length, Petal.Width)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# `mutate` - create new columns"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "iris %>%\n",
+    "    mutate(Sepal.Area = Sepal.Width * Sepal.Length,\n",
+    "          Petal.Area = Petal.Width * Petal.Length) %>%\n",
```

Comment:
Technically, this would only be the area if the sepals and petals were rectangular. I think the measured width and length is at the widest and longest point of the sepals and petals, so you are calculating the area of the minimum bounding box. Sorry for ruining your short column names =)